tools/build-tools/src/test/typeValidator/interfaceValidator.spec.ts

```diff
@@ -0,0 +1,391 @@
+            `
+            export interface ITestInterface {
+                newMethod?();
+            }
+            `;
+
+            let increment = checkIncrement(sourceOld, sourceNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // getters are treated the same as public methods
+        // adding/removing one is a breaking change
+        // TODO: getters currently unsupported in ts-morph but we don't use this/it's rare
+        it.skip("added a getter", () => {
+            const sourceOld =
+            `
+            export interface ITestInterface {
+            }
+            `;
+
+            const sourceNew =
+            `
+            export interface ITestInterface {
+                get testProperty(): string;
+            }
+            `;
+
+            let increment = checkIncrement(sourceOld, sourceNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // setters are treated the same as public methods
+        // adding/removing one is a breaking change
+        // TODO: setters currently unsupported in ts-morph but we don't use this/it's rare
+        it.skip("added a setter", () => {
+            const sourceOld =
+            `
+            export interface ITestInterface {
+            }
+            `;
+
+            const sourceNew =
+            `
+            export interface ITestInterface {
+                set setTestProperty(newTestProperty: string);
+            }
+            `;
+
+            let increment = checkIncrement(sourceOld, sourceNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // adding a new required method parameter is a breaking change
+        it("added a new required method param", () => {
+            const sourceOld =
+            `
+            export interface ITestInterface {
+                testMethod();
+            }
+            `;
+
+            const sourceNew =
+            `
+            export interface ITestInterface {
+                testMethod(newProp: boolean);
+            }
+            `;
+
+            let increment = checkIncrement(sourceOld, sourceNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+    });
+
+    describe("properties", () => {
+        // adding a readonly modifier is a breaking change
+        it.skip("added a readonly modifier", () => {
+            const sourceOld =
+            `
+            export interface ITestInterface {
+                myProp: number;
+            }
+            `;
+
+            const sourceNew =
+            `
+            export interface ITestInterface {
+                readonly myProp: number;
+            }
+            `;
+
+            let increment = checkIncrement(sourceOld, sourceNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // removing a readonly modifier is an incremental change
```

Comment:
Is this just being overly conservative?

Reply:
Oh, I guess any code reading the value wouldn't be expecting it to change.

Reply:
i think this case can be considered flexible in what behavior we want, but generally i think anything that changes the public API should get at least a minor increment